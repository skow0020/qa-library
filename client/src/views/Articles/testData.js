export const articles = {
  data: [
    {
      '_id': '5e220996d7ca7340bcfc9dd5',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Benny Schaden Jr.',
      'category': 'General',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Est quibusdam laboriosam ipsum cum est sunt numquam voluptas. Quas cupiditate facere vero.',
      'article_id': 1000
    },
    {
      '_id': '5e220996d7ca7340bcfc9dd6',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Joelle Mayer',
      'category': 'UI Automation',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Dolorum consequuntur suscipit impedit impedit repudiandae qui cupiditate quis rerum. Veritatis ab ab culpa praesentium necessitatibus vitae quasi.',
      'article_id': 1001
    },
    {
      '_id': '5e220996d7ca7340bcfc9dd7',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Leonardo Wunsch',
      'category': 'DeVops',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Quas eius non iure. Voluptatibus cupiditate recusandae libero repellendus velit ex.',
      'article_id': 1002
    }
  ]
};