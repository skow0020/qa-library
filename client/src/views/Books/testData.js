export const books = [
  {
    "backgroundImage": "https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg",
    "author": "Nova Goldner",
    "category": "API Automation",
    "language": "Ruby",
    "url": "https://tap-library-dev.herokuapp.com/tap-dashboard",
    "title": "How to sand a hippo",
    "pdf": "http://www.images.com/crazy.pdf",
    "body": "Est dolor ad. Provident deserunt aliquid sed provident minus fugit.",
    "book_id": 2002
  },
  {
    "backgroundImage": "https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg",
    "author": "Erin Medhurst",
    "category": "UI Automation",
    "language": "Swift",
    "url": "https://tap-library-dev.herokuapp.com/tap-dashboard",
    "title": "How to sand a hippo",
    "pdf": "http://www.images.com/crazy.pdf",
    "body": "Non mollitia dolores dolorum eos unde. Aut enim dolore nesciunt nemo tenetur et rem.",
    "book_id": 1001
  }
];