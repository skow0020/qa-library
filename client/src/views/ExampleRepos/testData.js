export const repos = [
  {
    'id': 198064850,
    'node_id': 'MDEwOlJlcG9zaXRvcnkxOTgwNjQ4NTA=',
    'name': 'acme-outreach',
    'full_name': 'skow0020/acme-outreach',
    'private': false,
    'owner': {
      'login': 'skow0020',
      'id': 12362090,
      'node_id': 'MDQ6VXNlcjEyMzYyMDkw',
      'avatar_url': 'https://avatars2.githubusercontent.com/u/12362090?v=4',
      'gravatar_id': '',
      'url': 'https://api.github.com/users/skow0020',
      'html_url': 'https://github.com/skow0020',
      'followers_url': 'https://api.github.com/users/skow0020/followers',
      'following_url': 'https://api.github.com/users/skow0020/following{/other_user}',
      'gists_url': 'https://api.github.com/users/skow0020/gists{/gist_id}',
      'starred_url': 'https://api.github.com/users/skow0020/starred{/owner}{/repo}',
      'subscriptions_url': 'https://api.github.com/users/skow0020/subscriptions',
      'organizations_url': 'https://api.github.com/users/skow0020/orgs',
      'repos_url': 'https://api.github.com/users/skow0020/repos',
      'events_url': 'https://api.github.com/users/skow0020/events{/privacy}',
      'received_events_url': 'https://api.github.com/users/skow0020/received_events',
      'type': 'User',
      'site_admin': false
    },
    'html_url': 'https://github.com/skow0020/acme-outreach',
    'description': null,
    'fork': false,
    'url': 'https://api.github.com/repos/skow0020/acme-outreach',
    'forks_url': 'https://api.github.com/repos/skow0020/acme-outreach/forks',
    'keys_url': 'https://api.github.com/repos/skow0020/acme-outreach/keys{/key_id}',
    'collaborators_url': 'https://api.github.com/repos/skow0020/acme-outreach/collaborators{/collaborator}',
    'teams_url': 'https://api.github.com/repos/skow0020/acme-outreach/teams',
    'hooks_url': 'https://api.github.com/repos/skow0020/acme-outreach/hooks',
    'issue_events_url': 'https://api.github.com/repos/skow0020/acme-outreach/issues/events{/number}',
    'events_url': 'https://api.github.com/repos/skow0020/acme-outreach/events',
    'assignees_url': 'https://api.github.com/repos/skow0020/acme-outreach/assignees{/user}',
    'branches_url': 'https://api.github.com/repos/skow0020/acme-outreach/branches{/branch}',
    'tags_url': 'https://api.github.com/repos/skow0020/acme-outreach/tags',
    'blobs_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/blobs{/sha}',
    'git_tags_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/tags{/sha}',
    'git_refs_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/refs{/sha}',
    'trees_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/trees{/sha}',
    'statuses_url': 'https://api.github.com/repos/skow0020/acme-outreach/statuses/{sha}',
    'languages_url': 'https://api.github.com/repos/skow0020/acme-outreach/languages',
    'stargazers_url': 'https://api.github.com/repos/skow0020/acme-outreach/stargazers',
    'contributors_url': 'https://api.github.com/repos/skow0020/acme-outreach/contributors',
    'subscribers_url': 'https://api.github.com/repos/skow0020/acme-outreach/subscribers',
    'subscription_url': 'https://api.github.com/repos/skow0020/acme-outreach/subscription',
    'commits_url': 'https://api.github.com/repos/skow0020/acme-outreach/commits{/sha}',
    'git_commits_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/commits{/sha}',
    'comments_url': 'https://api.github.com/repos/skow0020/acme-outreach/comments{/number}',
    'issue_comment_url': 'https://api.github.com/repos/skow0020/acme-outreach/issues/comments{/number}',
    'contents_url': 'https://api.github.com/repos/skow0020/acme-outreach/contents/{+path}',
    'compare_url': 'https://api.github.com/repos/skow0020/acme-outreach/compare/{base}...{head}',
    'merges_url': 'https://api.github.com/repos/skow0020/acme-outreach/merges',
    'archive_url': 'https://api.github.com/repos/skow0020/acme-outreach/{archive_format}{/ref}',
    'downloads_url': 'https://api.github.com/repos/skow0020/acme-outreach/downloads',
    'issues_url': 'https://api.github.com/repos/skow0020/acme-outreach/issues{/number}',
    'pulls_url': 'https://api.github.com/repos/skow0020/acme-outreach/pulls{/number}',
    'milestones_url': 'https://api.github.com/repos/skow0020/acme-outreach/milestones{/number}',
    'notifications_url': 'https://api.github.com/repos/skow0020/acme-outreach/notifications{?since,all,participating}',
    'labels_url': 'https://api.github.com/repos/skow0020/acme-outreach/labels{/name}',
    'releases_url': 'https://api.github.com/repos/skow0020/acme-outreach/releases{/id}',
    'deployments_url': 'https://api.github.com/repos/skow0020/acme-outreach/deployments',
    'created_at': '2019-07-21T14:13:49Z',
    'updated_at': '2019-07-21T14:37:03Z',
    'pushed_at': '2019-07-21T14:37:01Z',
    'git_url': 'git://github.com/skow0020/acme-outreach.git',
    'ssh_url': 'git@github.com:skow0020/acme-outreach.git',
    'clone_url': 'https://github.com/skow0020/acme-outreach.git',
    'svn_url': 'https://github.com/skow0020/acme-outreach',
    'homepage': null,
    'size': 164,
    'stargazers_count': 0,
    'watchers_count': 0,
    'language': 'JavaScript',
    'has_issues': true,
    'has_projects': true,
    'has_downloads': true,
    'has_wiki': true,
    'has_pages': false,
    'forks_count': 0,
    'mirror_url': null,
    'archived': false,
    'disabled': false,
    'open_issues_count': 0,
    'license': null,
    'forks': 0,
    'open_issues': 0,
    'watchers': 0,
    'default_branch': 'master'
  },
  {
    'id': 85446048,
    'node_id': 'MDEwOlJlcG9zaXRvcnk4NTQ0NjA0OA==',
    'name': 'Appium-Tutorial',
    'full_name': 'skow0020/Appium-Tutorial',
    'private': false,
    'owner': {
      'login': 'skow0020',
      'id': 12362090,
      'node_id': 'MDQ6VXNlcjEyMzYyMDkw',
      'avatar_url': 'https://avatars2.githubusercontent.com/u/12362090?v=4',
      'gravatar_id': '',
      'url': 'https://api.github.com/users/skow0020',
      'html_url': 'https://github.com/skow0020',
      'followers_url': 'https://api.github.com/users/skow0020/followers',
      'following_url': 'https://api.github.com/users/skow0020/following{/other_user}',
      'gists_url': 'https://api.github.com/users/skow0020/gists{/gist_id}',
      'starred_url': 'https://api.github.com/users/skow0020/starred{/owner}{/repo}',
      'subscriptions_url': 'https://api.github.com/users/skow0020/subscriptions',
      'organizations_url': 'https://api.github.com/users/skow0020/orgs',
      'repos_url': 'https://api.github.com/users/skow0020/repos',
      'events_url': 'https://api.github.com/users/skow0020/events{/privacy}',
      'received_events_url': 'https://api.github.com/users/skow0020/received_events',
      'type': 'User',
      'site_admin': false
    },
    'html_url': 'https://github.com/skow0020/Appium-Tutorial',
    'description': 'Personal project for learning Appium - In Progress',
    'fork': false,
    'url': 'https://api.github.com/repos/skow0020/Appium-Tutorial',
    'forks_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/forks',
    'keys_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/keys{/key_id}',
    'collaborators_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/collaborators{/collaborator}',
    'teams_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/teams',
    'hooks_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/hooks',
    'issue_events_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/issues/events{/number}',
    'events_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/events',
    'assignees_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/assignees{/user}',
    'branches_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/branches{/branch}',
    'tags_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/tags',
    'blobs_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/blobs{/sha}',
    'git_tags_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/tags{/sha}',
    'git_refs_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/refs{/sha}',
    'trees_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/trees{/sha}',
    'statuses_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/statuses/{sha}',
    'languages_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/languages',
    'stargazers_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/stargazers',
    'contributors_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/contributors',
    'subscribers_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/subscribers',
    'subscription_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/subscription',
    'commits_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/commits{/sha}',
    'git_commits_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/commits{/sha}',
    'comments_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/comments{/number}',
    'issue_comment_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/issues/comments{/number}',
    'contents_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/contents/{+path}',
    'compare_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/compare/{base}...{head}',
    'merges_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/merges',
    'archive_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/{archive_format}{/ref}',
    'downloads_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/downloads',
    'issues_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/issues{/number}',
    'pulls_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/pulls{/number}',
    'milestones_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/milestones{/number}',
    'notifications_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/notifications{?since,all,participating}',
    'labels_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/labels{/name}',
    'releases_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/releases{/id}',
    'deployments_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/deployments',
    'created_at': '2017-03-19T02:36:41Z',
    'updated_at': '2017-03-20T14:03:03Z',
    'pushed_at': '2017-03-31T01:34:23Z',
    'git_url': 'git://github.com/skow0020/Appium-Tutorial.git',
    'ssh_url': 'git@github.com:skow0020/Appium-Tutorial.git',
    'clone_url': 'https://github.com/skow0020/Appium-Tutorial.git',
    'svn_url': 'https://github.com/skow0020/Appium-Tutorial',
    'homepage': '',
    'size': 126,
    'stargazers_count': 0,
    'watchers_count': 0,
    'language': 'Java',
    'has_issues': true,
    'has_projects': true,
    'has_downloads': true,
    'has_wiki': true,
    'has_pages': false,
    'forks_count': 0,
    'mirror_url': null,
    'archived': false,
    'disabled': false,
    'open_issues_count': 0,
    'license': null,
    'forks': 0,
    'open_issues': 0,
    'watchers': 0,
    'default_branch': 'master'
  }
];
