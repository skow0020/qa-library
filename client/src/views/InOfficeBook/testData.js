export const inOfficeBook = {
  data: [{
    'office_book_id': '1000',
    'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
    'author': 'Nova Goldner',
    'category': 'API Automation',
    'url': 'https://tap-library-dev.herokuapp.com/tap-dashboard',
    'title': 'How to sand a hippo',
    'body': 'Est dolor ad. Provident deserunt aliquid sed provident minus fugit.',
    'totalCopies': 6,
    'copiesCheckedOut': 2,
    'users': ['stanislov342', 'georgio232']
  }]
};