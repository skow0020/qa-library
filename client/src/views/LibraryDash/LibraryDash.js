import AlertModal, { showAlert } from "components/common/AlertModal";
import {
  Badge,
  Button,
  Card,
  CardBody,
  CardFooter,
  Col,
  Container,
  FormGroup,
  Row
} from "shards-react";

import CategoriesSelection from "components/common/CategoriesSelection";
import { Link } from "react-router-dom";
import LoadError from "components/common/LoadError";
import Loading from "components/common/Loading";
import PageTitle from "components/common/PageTitle";
import React from "react";
import { Store } from "../../flux";
import axios from "axios";
import { getCategoryTheme } from "utils/util";
import queryString from 'query-string';

class LibraryDash extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: null,
      avatarUrl: null,
      isLoggedIn: false,
      books: [],
      isLoading: false
    };

    this.handleFilterChange = this.handleFilterChange.bind(this);
    this.handleCheckout = this.handleCheckout.bind(this);
    this.handleCheckIn = this.handleCheckIn.bind(this);
  }

  componentDidMount() {
    if (this.props.hasOwnProperty('location') && this.props.location.search) {
      const values = queryString.parse(this.props.location.search);
      if (values.user) {
        Store.user = values.user;
        Store.avatarUrl = values.avatar_url;
        this.setState({
          user: values.user,
          avatarUrl: values.avatar_url,
          isLoggedIn: true
        });
        this.getBookList();
      }
    } else if (Store.user) {
      this.setState({
        user: Store.user,
        avatarUrl: Store.avatarUrl,
        isLoggedIn: true
      });
      this.getBookList();
    }
  }

  getBookList() {
    const categoryFilter = this.state.category ? `category=${this.state.category}` : '';
    const filter = categoryFilter;

    this.setState({ isLoading: true });
    fetch(`/api/officeLibraryBooks?${filter}`)
      .then(response => response.json())
      .then(
        data => this.setState({ books: data.data, isLoading: false }),
        error => this.setState({ error, isLoading: false })
      );
  }

  handleFilterChange(e) {
    this.setState({
      category: e.target.value
    }, () => this.getBookList());
  }

  handleCheckout(book_id) {
    const checkoutBook = {
      office_book_id: book_id,
      user: Store.user
    };
    axios.patch("/api/officeLibraryBooks/incrementCopiesCheckedOut", checkoutBook)
      .then(
        data => {
          if (!data.data.success) this.showNotifier(`Unable to check out book: ${data.data.error}`);
          else this.getBookList();
        },
        error => this.showNotifier(`Unable to checkout book: ${error}`)
      );
  }

  handleCheckIn(book_id) {
    const checkinBook = {
      office_book_id: book_id,
      user: Store.user
    };
    axios.patch("/api/officeLibraryBooks/decrementCopiesCheckedOut", checkinBook)
      .then(
        data => {
          if (!data.data.success) this.showNotifier(`Unable to check in book: ${data.data.error}`);
          else this.getBookList();
        },
        error => this.showNotifier(`Unable to checkin book: ${error}`)
      );
  }

  showNotifier = (message) => {
    showAlert({ message });
  }

  render() {
    const { category, isLoggedIn, isLoading, error, books } = this.state;

    if (isLoading) return <Loading />;

    if (error) return <LoadError error="Library books failed to load" />;

    let login;
    if (!isLoggedIn) {
      login = <a href="/login" aria-label="Login to Github">
        <Button id="login-button" type="button" className="btn btn-success btn-lg">Log In with Github</Button>
      </a>;
    } else login = (
      <div>
        <div>
          <span className="d-none d-md-inline-block px-3">{this.state.user}</span>
          <img className="user-avatar rounded-circle mr-2" src={this.state.avatarUrl} alt="User Avatar" width="70" />
        </div>
      </div>
    );

    let categoryDD;
    if (isLoggedIn) {
      categoryDD = (
        <FormGroup id='filtering-form'>
          <CategoriesSelection id="category" value={category} onChange={this.handleFilterChange} />
        </FormGroup>
      );
    }

    return (
      <Container fluid className="main-content-container px-3">
        <AlertModal />
        <Row noGutters className="form-inline py-4">
          <PageTitle sm="8" title="In-Office Library" className="text-sm-left" />
          {login}
          {categoryDD}
        </Row>
        <Row>
          {books.map((book, idx) => (
            <Col lg="4" md="6" sm="12" className="mb-4" key={idx}>
              <Card small id={`book-card-${idx}`} className="card-book card-post--1">
                <Link id={`book-${book.office_book_id}`} to={`/officeBook/${book.office_book_id}`}>
                  <div
                    className="card-post__image"
                    style={{ backgroundImage: `url('${book.backgroundImage}')` }}
                  />
                </Link>
                <Badge pill className={`card-post__category bg-${getCategoryTheme(book.category)}`}>
                  {book.category}
                </Badge>
                <Link id={`book-body-${book.office_book_id}`} to={`/officeBook/${book.office_book_id}`}>
                  <CardBody>
                    <h5 className="card-title">
                      <p className="text-fiord-blue">
                        {book.title}
                      </p>
                    </h5>
                  </CardBody>
                </Link>
                <CardFooter className="text-muted border-top py-3">
                  <Row>
                    <span className="d-inline-block px-3">
                      By{" "}{book.author} | {book.totalCopies - book.copiesCheckedOut}{" "}Available
                    </span>
                  </Row>
                  <div>
                    <Button type="button" className={`checkout-${book.office_book_id} btn btn-success`} onClick={() => this.handleCheckout(book.office_book_id)}>
                      Check out
                    </Button>
                    <span className="d-inline-block px-3">
                      {" "}
                    </span>
                    <Button type="button" className={`checkin-${book.office_book_id} btn btn-success pull-right`} onClick={() => this.handleCheckIn(book.office_book_id)}>
                      Check in
                    </Button>
                  </div>
                </CardFooter>
              </Card>
            </Col>
          ))}
        </Row>
      </Container>
    );
  }
}
export default LibraryDash;
