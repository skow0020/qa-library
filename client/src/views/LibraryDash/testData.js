export const inOfficeBooks = {
  data: [
      {
        "backgroundImage": "https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg",
        "author": "Nova Goldner",
        "category": "API Automation",
        "url": "https://tap-library-dev.herokuapp.com/tap-dashboard",
        "title": "How to sand a hippo",
        "body": "Est dolor ad. Provident deserunt aliquid sed provident minus fugit.",
        "totalCopies": 6,
        "copiesCheckedOut": 2,
        "users": ["stanislov342", "georgio232"]
      },
      {
        "backgroundImage": "https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg",
        "author": "Erin Medhurst",
        "category": "UI Automation",
        "url": "https://tap-library-dev.herokuapp.com/tap-dashboard",
        "title": "How to sand a hippo",
        "body": "Non mollitia dolores dolorum eos unde. Aut enim dolore nesciunt nemo tenetur et rem.",
        "totalCopies": 2,
        "copiesCheckedOut": 1,
        "users": ["georgio232"]
      }
    ]
};