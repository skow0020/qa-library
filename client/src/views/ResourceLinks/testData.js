export const resourceLinks = {
  data: [
    {
      '_id': '5e220997d7ca7340bcfc9ded',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'category': 'General',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Corporis laborum assumenda non qui. Soluta facilis dolore doloremque enim facere a accusantium.',
      'res_id': 1000
    },
    {
      '_id': '5e220997d7ca7340bcfc9dee',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'category': 'UI Automation',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Voluptatem qui dolore veritatis ullam tempore eos. Deserunt autem nulla praesentium magnam dolorem.',
      'res_id': 1001
    },
    {
      '_id': '5e220997d7ca7340bcfc9def',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'category': 'DeVops',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Illo qui consequatur eum. Tenetur sed vel est quod.',
      'res_id': 1002
    }
  ]
};