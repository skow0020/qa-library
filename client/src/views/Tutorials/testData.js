export const tutorials = {
  data: [
    {
      '_id': '5e220997d7ca7340bcfc9df6',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Itzel Wiza DVM',
      'category': 'General',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Ut nihil optio est est vel assumenda cum totam sunt. Mollitia incidunt nulla aperiam et aut nobis.',
      'tut_id': 1000
    },
    {
      '_id': '5e220997d7ca7340bcfc9df7',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Annalise Fahey',
      'category': 'UI Automation',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Et temporibus animi eius et inventore minus sunt voluptatum est. Eveniet consectetur voluptatem.',
      'tut_id': 1001
    },
    {
      '_id': '5e220997d7ca7340bcfc9df8',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Shayne Lubowitz',
      'category': 'DeVops',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Veniam labore vel unde est esse. Provident dolores laudantium consequatur.',
      'tut_id': 1002
    }
  ]
};