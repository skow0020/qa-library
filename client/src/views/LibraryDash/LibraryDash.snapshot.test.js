import LibraryDash from './LibraryDash';
import React from 'react';
import { inOfficeBooks } from './testData';
import { shallow, mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import { BrowserRouter as Router } from 'react-router-dom';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('LibraryDash Snapshot Tests', () => {
  const state = {
    isLoggedIn: true,
    books: inOfficeBooks,
    isLoading: false
  };

  test('LibraryDash loading snapshot', () => {
    const wrapper = shallow(<LibraryDash />);

    wrapper.setState({ isLoading: true });

    expect(wrapper).toMatchSnapshot();
  });

  test('LibraryDash snapshot', () => {

    const wrapper = mount(
      <Router>
        <LibraryDash />
      </Router>
    );

    wrapper.setState(state);

    expect(wrapper).toMatchSnapshot();
  });
});