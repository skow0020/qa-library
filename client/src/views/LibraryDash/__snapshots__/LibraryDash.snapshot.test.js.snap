// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibraryDash Snapshot Tests LibraryDash loading snapshot 1`] = `<Loading />`;

exports[`LibraryDash Snapshot Tests LibraryDash snapshot 1`] = `
<div
  className="main-content-container px-3 container-fluid"
>
  <div />
  <div
    className="form-inline py-4 no-gutters row"
  >
    <div
      className="text-sm-left text-center text-md-left mb-sm-0 col-12 col-sm-8"
    >
      <span
        className="text-uppercase page-subtitle"
      />
      <h3
        className="page-title"
      >
        In-Office Library
      </h3>
    </div>
    <a
      aria-label="Login to Github"
      href="/login"
    >
      <button
        className="btn btn-success btn-lg btn btn-primary"
        id="login-button"
        onClick={[Function]}
        type="button"
      >
        Log In with Github
      </button>
    </a>
  </div>
  <div
    className="row"
  />
</div>
`;
